MovieCave - MVVM/MovieCave/MovieCave/VIew/TVSeriesView/TVSeriesViewController.swift

```diff
@@ -0,0 +1,106 @@
+//
+//  AddMovieViewController.swift
+//  MovieCave
+//
+//  Created by Admin on 21.09.23.
+//
+
+import UIKit
+import Combine
+
+class TVSeriesViewController: UIViewController, SpinnerProtocol {
+
+    //MARK: - IBOutlets
+    @IBOutlet private weak var tvSeriesView: ReusableListViewProtocol!
+    
+    //MARK: - Properties
+    var viewModel: TVSeriesViewModelProtocol?
+    var spinnerView: UIView?
+    private var popUp: PopUpView!
+    private var cancellables = [AnyCancellable]()
+
+    //MARK: - Lifecycle Methods
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        navigationController?.setNavigationBarHidden(true, animated: false)
+        tvSeriesView.filterButtons(buttonTitles: [Constants.popularTVSeriesFilterButton,
+                                                  Constants.airingTodayTVSeriesFilterButton,
+                                                  Constants.onTheAirTVSeriesFilterButton,
+                                                  Constants.topRatedTVSeriesFilterButton])
+        setUpBinders()
+        setUpDelegates()
+    }
+    
+    //MARK: - Private Methods
+    private func setUpBinders() {
+        viewModel?.resetPosition.sink { [weak self] makeReset in
+            guard let self,
+                  makeReset else { return }
+            
+            self.showSpinner()
+            DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) {
+                self.tvSeriesView.resetCollectionViewPosition(with: 0, on: 0, animation: true)
+            }
+            
+        }.store(in: &cancellables)
+        
+        viewModel?.fetchingDataSuccession.sink { [weak self] success in
+            guard let self,
+                  success else { return }
+            
+            self.removeSpinner()
+            DispatchQueue.main.async {
```

Comment:
Are you sure you need this line?

Reply:
#
<img width="1041" alt="Screenshot 2024-02-07 at 15 08 08" src="https://github.com/oniqotlqvo22/Projects/assets/70860946/72ee5e53-05af-438a-9994-ccea56bac705">

Reply:
line 208
collectionView.reloadSections must be called only in the main thread